src/pages/Login/Login.container.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+import Login from './Login.view';
+
+const LoginContainer = () => {
+  const isAuthenticated = localStorage.getItem('auth-token');
+  const [username, setUsername] = React.useState('');
+  const [password, setPassword] = React.useState('');
+  const [showPassword, setShowPassword] = React.useState(false);
```

Comment:
Agree. I did like this due to two reasons - 

1. Using useState, means we are maintaining state outside of function.
2. I was not able to test setter function provided by useState.

Reply:
So generally we don't directly test the setter functions. Instead we test the effect it does on the component.
For example `setUsername` changes the `username` which is used like `value={username}` in the input field of Username. So instead of testing the setter function we can test value of username input field after firing the onchange event on the username field.

Reply:
Yes, I think you are correct.